src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -96,23 +96,19 @@
         earliestSinceDate = null;
         progressTracker = new ProgressTracker(configs.size());
 
-        cloneAndAnalyzeRepos(configs, outputPath);
+        List<Path> reportFoldersAndFiles = cloneAndAnalyzeRepos(configs, outputPath);
 
         Date reportSinceDate = (cliSinceDate.equals(SinceDateArgumentType.ARBITRARY_FIRST_COMMIT_DATE))
                 ? earliestSinceDate : cliSinceDate;
 
-        FileUtil.writeJsonFile(
+        Optional<Path> summaryPath = FileUtil.writeJsonFile(
                 new SummaryJson(configs, generationDate, reportSinceDate, untilDate, isSinceDateProvided,
                         isUntilDateProvided, RepoSense.getVersion(), ErrorSummary.getInstance().getErrorList()),
-                getSummaryResultPath(outputPath));
+                        getSummaryResultPath(outputPath));
```

Comment:
the `getSummaryResultPath(...)` should be an argument of `FileUtil.writeJsonFile()` and not the `SummaryJson` constructor, so it should be 8 spaces relative to line 104.


```java
Optional<Path> summaryPath = FileUtil.writeJsonFile(
        new SummaryJson(configs, generationDate, reportSinceDate, untilDate, isSinceDateProvided,
                isUntilDateProvided, RepoSense.getVersion(), ErrorSummary.getInstance().getErrorList()),
        getSummaryResultPath(outputPath));
```

Reply:
Done.